python/ray/tune/suggest/hebo.py

```diff
@@ -219,11 +228,12 @@ def set_search_properties(self, metric: Optional[str], mode: Optional[str],
                               config: Dict, **spec) -> bool:
         if self._opt:
             return False
-        space = self.convert_search_space(config)
-        self._space = space
-        if metric:
+        if not self._space:
+            space = self.convert_search_space(config)
+            self._space = space
+        if not self._metric:
             self._metric = metric
-        if mode:
+        if not self._mode:
```

Comment:
For all other searchers, we have `if mode:` here, in order to always overwrite the mode if passed to `tune.run()`. Should we keep this here, too?

Reply:
You are right, it should be consistent. Changed